AlphaWallet/Settings/ViewControllers/SettingsViewController.swift

```diff
@@ -107,6 +107,30 @@ class SettingsViewController: UIViewController {
         }
     }
 
+    private func configureChangeWalletCellWithResolvedESN(_ row: SettingsWalletRow, cell: SettingTableViewCell) {
+        let viewModel: SettingTableViewCellViewModel = .init(
+            titleText: row.title,
+            subTitleText: account.address.eip55String,
+            icon: row.icon
+        )
+
+        let serverToResolveEns = RPCServer.main
+        let address = account.address
+
+        ENSReverseLookupCoordinator(server: serverToResolveEns).getENSNameFromResolver(forAddress: address) { result in
+            guard let ensName = result.value else { return }
+
+            let viewModel: SettingTableViewCellViewModel = .init(
+                titleText: row.title,
+                subTitleText: ensName,
+                icon: row.icon
+            )
+            cell.configure(viewModel: viewModel)
+        }
+
+        cell.configure(viewModel: viewModel)
```

Comment:
The second `cell.configure(viewModel: viewModel)` has to be moved up, to before calling `getENSNameFromResolver()` because the latter can return a cached result immediately. You can see the problem by doing this:

1. Switch to Settings tab
2. Wait for ENS reverse resolution to complete and ENS name shown
3. Tap "Change / Add Wallet"
4. Tap "Done".
5. Observe: it shows the address and wouldn't show ENS name even after waiting for a while. Expected: Show ENS name (and right away because of caching).